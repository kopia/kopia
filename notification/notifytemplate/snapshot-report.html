Subject: {{.EventArgs.OverallStatus}} on {{.Hostname}}

<!doctype html>
<html>
<head>
<style type="text/css">
    table {
        width: 100%;
        border-collapse: collapse;
    }

    thead tr {
        background-color: #f2f2f2;
    }

    th, td {
        border: 1px solid black;
        padding: 2px;
        text-align: left;
    }

    th {
        background-color: #f2f2f2;
    }

    code {
        background-color: #f2f2f2;
        padding: 2px;
    }

    span.path {
        font-family: monospace;
        color: #344652;
        font-weight: bold;
    }

    span.increase {
        color: green;
        font-style: italic;
    }

    span.decrease {
        color: red;
        font-style: italic;
    }

    tr.snapshotstatus-fatal {
        background-color: #fde9e4;
    }

    tr.snapshotstatus-error {
        background-color: #fcffba;
    }

    tr.snapshotstatus-incomplete {
        background-color: #8a8c7e;
    }
</style>
</head>
<body>
<table border="1">
<thead>
    <tr>
        <th>Source</th>
        <th>Started</th>
        <th>Duration</th>
        <th>Total Size</th>
        <th>Total Files</th>
        <th>Total Directories</th>
    </tr>
</thead>
{{ range .EventArgs.Snapshots | sortSnapshotManifestsByName}}
<tr class="snapshotstatus-{{ .StatusCode }}">
<td><span class="path">{{ .Manifest.Source.Path }}</span></td>
<td>{{ .StartTimestamp | formatTime }}</td>
<td>{{ .Duration }}</td>
<td>{{ .TotalSize | bytes }}{{ .TotalSizeDelta | bytesDeltaHTML }}</td>
<td>{{ .TotalFiles | formatCount }}{{ .TotalFilesDelta | countDeltaHTML }}</td>
<td>{{ .TotalDirs | formatCount }}{{ .TotalDirsDelta | countDeltaHTML }}</td>
</tr>

{{ if .Error }}
<tr class="snapshotstatus-{{ .StatusCode }}">
    <td colspan="6">
        <b style="color:red">Error:</b> {{ .Error }}
    </td>
</tr>
{{ end }}

{{ if .Manifest.RootEntry }}
{{ if .Manifest.RootEntry.DirSummary }}
{{ if .Manifest.RootEntry.DirSummary.FailedEntries }}
<tr class="snapshotstatus-{{ .StatusCode }}">
    <td colspan="6">
        <b style="color:red">Failed Entries:</b>
        <ul>
        {{ range .Manifest.RootEntry.DirSummary.FailedEntries }}
        <li><span class="path">{{.EntryPath}}</span>: {{.Error}}</li>
        {{ end }}
        </ul>
    </td>
</tr>
{{ end }}
{{ end }}
{{ end }}

{{ end }}
</table>

<p>Generated at {{ .EventTime | formatTime }} by <a href="https://kopia.io">Kopia {{ .KopiaBuildVersion }}</a>.</p>

</body>
</html>
