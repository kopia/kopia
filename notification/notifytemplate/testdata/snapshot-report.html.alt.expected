Subject: Failed to create 1 of 4 snapshots on some-host

<!doctype html>
<html>
<head>
<style type="text/css">
    table {
        width: 100%;
        border-collapse: collapse;
    }

    thead tr {
        background-color: #f2f2f2;
    }

    th, td {
        border: 1px solid black;
        padding: 2px;
        text-align: left;
    }

    th {
        background-color: #f2f2f2;
    }

    code {
        background-color: #f2f2f2;
        padding: 2px;
    }

    span.path {
        font-family: monospace;
        color: #344652;
        font-weight: bold;
    }

    span.increase {
        color: green;
        font-style: italic;
    }

    span.decrease {
        color: red;
        font-style: italic;
    }

    tr.snapshotstatus-fatal {
        background-color: #fde9e4;
    }

    tr.snapshotstatus-error {
        background-color: #fcffba;
    }

    tr.snapshotstatus-incomplete {
        background-color: #8a8c7e;
    }
</style>
</head>
<body>
<table border="1">
<thead>
    <tr>
        <th>Source</th>
        <th>Started</th>
        <th>Duration</th>
        <th>Total Size</th>
        <th>Total Files</th>
        <th>Total Directories</th>
    </tr>
</thead>

<tr class="snapshotstatus-fatal">
<td><span class="path">/some/other/path</span></td>
<td>Wed, 31 Dec 1969 16:00:00 PST</td>
<td>0s</td>
<td>0 B</td>
<td>0</td>
<td>0</td>
</tr>


<tr class="snapshotstatus-fatal">
    <td colspan="6">
        <b style="color:red">Error:</b> some top-level error
    </td>
</tr>





<tr class="snapshotstatus-success">
<td><span class="path">/some/path</span></td>
<td>Wed, 01 Jan 2020 19:04:05 PST</td>
<td>1.1s</td>
<td>456 B <span class='increase'>(&#x2191; 56 B)</span></td>
<td>123 <span class='increase'>(&#x2191; 23)</span></td>
<td>33 <span class='increase'>(&#x2191; 3)</span></td>
</tr>






<tr class="snapshotstatus-success">
    <td colspan="6">
        <b style="color:red">Failed Entries:</b>
        <ul>
        
        <li><span class="path">/some/path</span>: some error</li>
        
        <li><span class="path">/some/path2</span>: some error</li>
        
        </ul>
    </td>
</tr>





<tr class="snapshotstatus-success">
<td><span class="path">/some/path</span></td>
<td>Wed, 01 Jan 2020 19:04:05 PST</td>
<td>1.1s</td>
<td>456 B <span class='decrease'>(&#x2193; 44 B)</span></td>
<td>123 <span class='decrease'>(&#x2193; 77)</span></td>
<td>33 <span class='decrease'>(&#x2193; 7)</span></td>
</tr>






<tr class="snapshotstatus-success">
    <td colspan="6">
        <b style="color:red">Failed Entries:</b>
        <ul>
        
        <li><span class="path">/some/path</span>: some error</li>
        
        <li><span class="path">/some/path2</span>: some error</li>
        
        </ul>
    </td>
</tr>





<tr class="snapshotstatus-success">
<td><span class="path">/some/path2</span></td>
<td>Wed, 01 Jan 2020 19:04:05 PST</td>
<td>1.1s</td>
<td>456 B</td>
<td>123</td>
<td>33</td>
</tr>






<tr class="snapshotstatus-success">
    <td colspan="6">
        <b style="color:red">Failed Entries:</b>
        <ul>
        
        <li><span class="path">/some/path</span>: some error</li>
        
        <li><span class="path">/some/path2</span>: some error</li>
        
        </ul>
    </td>
</tr>





</table>

<p>Generated at Wed, 01 Jan 2020 19:04:05 PST by <a href="https://kopia.io">Kopia v0-unofficial</a>.</p>

</body>
</html>
